v2/health.go

```diff
@@ -0,0 +1,127 @@
+package shuttle
+
+import (
+	"context"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/messaging/azservicebus"
+	"github.com/Azure/go-shuttle/v2/metrics/processor"
+	"github.com/Azure/go-shuttle/v2/metrics/sender"
+)
+
+type HealthCheckFunc func(ctx context.Context, namespace string, client *azservicebus.Client) error
+
+// HealthChecker performs periodic health checks on the Service Bus Senders and Receivers.
+// It uses azservicebus.Sender.NewMessageBatch() and azservicebus.Receiver.PeekMessages() to perform the health checks.
+type HealthChecker struct {
+	// clients is a map of namespace name to azservicebus.Client.
+	clients map[string]*azservicebus.Client
+	// entity is the name of the queue or topic.
+	entity string
+	// subscription is the name of the subscription. Leave empty for queues and senders.
+	subscription string
```

Comment:
if the subscription is optional most of the time (only for receiver, and only for topics), should this be in the option structs too?
How does this work for topics anyways, we can't ensure that all subscribers understand this "protocol" so we cannot safely send a heathcheck message onto the topic. all subscriptions will receive it.